processes/procstat.go

```diff
@@ -130,22 +143,40 @@ func (psc *procStatsCollector) GetStats(procPath string) (map[string][]Proc, err
 				vmData = pStatus["VmData"] * 1024
 				vmCode = (pStatus["VmExe"] + pStatus["VmLib"]) * 1024
 			}
+
+			procStatFields := strings.Fields(string(procStat))
+			if len(procStatFields) < 3 {
+				return nil, fmt.Errorf("Cannot retrieve process state")
+			}
+			procState := procStatFields[2]
+
 			// TODO: gather task status data /proc/<pid>/task
```

Comment:
Yes, it is still valid, just TODO for new metrics gathered from "task" subdirectory in procfs.